tests/test_http_client.py

```diff
@@ -993,3 +1014,342 @@ def test_encode_array(self):
+        client = self.REQUEST_CLIENT(verify_ssl_certs=True)
+        return client.request_stream_with_retries_async(
+            method, url, headers, post_data
+        )
+
+    @pytest.fixture
+    def mock_response(self):
+        def mock_response(mock, body, code):
+            raise NotImplementedError(
+                "You must implement this in your test subclass"
+            )
+
+        return mock_response
+
+    @pytest.fixture
+    def mock_error(self):
+        def mock_error(mock, error):
+            raise NotImplementedError(
+                "You must implement this in your test subclass"
+            )
+
+        return mock_error
+
+    @pytest.fixture
+    def check_call(self):
+        def check_call(
+            mock, method, abs_url, headers, params, is_streaming=False
+        ):
+            raise NotImplementedError(
+                "You must implement this in your test subclass"
+            )
+
+        return check_call
+
+
+class TestHTTPXClient(StripeClientTestCase, ClientTestBaseAsync):
+    REQUEST_CLIENT: Type[_http_client.HTTPXClient] = _http_client.HTTPXClient
+
+    @pytest.fixture
+    def mock_response(self, mocker, request_mock):
+        def mock_response(body={}, code=200):
+            result = mocker.Mock()
+            result.content = body
+            result.status_code = code
+            result.headers = {}
+
+            async def do(*args, **kwargs):
+                return result
+
+            async_mock = AsyncMock(side_effect=do)
+
+            request_mock.AsyncClient().request = async_mock
+            return result
+
+        return mock_response
+
+    @pytest.fixture
+    def mock_error(self, mocker, request_mock):
+        def mock_error(mock):
+            # The first kind of request exceptions we catch
+            mock.exceptions.SSLError = Exception
+            request_mock.AsyncClient().request.side_effect = (
+                mock.exceptions.SSLError()
+            )
+
+        return mock_error
+
+    @pytest.fixture
+    def check_call(self, request_mock, mocker):
+        def check_call(
+            mock,
+            method,
+            url,
+            post_data,
+            headers,
+            is_streaming=False,
+            timeout=80,
+            times=None,
+        ):
+            times = times or 1
+            args = (method, url)
+            kwargs = {
+                "headers": headers,
+                "data": post_data,
+                "timeout": timeout,
+                "proxies": {"http": "http://slap/", "https": "http://slap/"},
+            }
+
+            if is_streaming:
+                kwargs["stream"] = True
+
+            calls = [mocker.call(*args, **kwargs) for _ in range(times)]
+            request_mock.AsyncClient().request.assert_has_calls(calls)
+
+        return check_call
+
+    async def make_request_async(
+        self, method, url, headers, post_data, timeout=80
+    ):
+        client = self.REQUEST_CLIENT(
```

Comment:
Why is this client-specific and not in the base test class?

Reply:
Oh, the reason turns out to be that `timeout` is not in the base class. Per our other discussion, I will fix this, but separately.